staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -152,6 +152,8 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 
 	clientConfig := &restclient.Config{}
 	clientConfig.Host = configClusterInfo.Server
+	clientConfig.BearerToken = configAuthInfo.Token
```

Comment:
This is done as part of the getUserIdentificationPartialConfig call below, but is only set for TLS connections. Are you using a non-TLS hostname?

/hold

Reply:
I think I am doing it for non-TLS hostname. Is that wrong approach?

Reply:
client-go only supports sending credentials over TLS connections